include/zephyr/linker/common-rom/common-rom-misc.ld

```diff
@@ -39,6 +39,10 @@
 	ITERABLE_SECTION_ROM(zbus_channel_observation, 4)
 #endif /* CONFIG_ZBUS */
 
+#if defined(CONFIG_MCUX_FLEXIO)
```

Comment:
I'm concerned about defining this `ITERABLE_SECTION_ROM` within the `common-rom-misc.ld` file, this location feels too generic. I'd rather see this macro placed in a snippet file of some kind within the `drivers/misc/mcux_flexio` folder, and included using `zephyr_linker_sources` within the FlexIO CMakeLists file, similar to the way Zephyr recommends applications use iterable sections: https://docs.zephyrproject.org/latest/kernel/iterable_sections/index.html#usage

Reply:
Another thought here- what are we gaining by using `STRUCT_SECTION_FOREACH` to initialize child FlexIO devices? Could we instead have each child FlexIO device define its own `struct mcux_flexio_child`, and call a function to "attach" to the parent FlexIO controller (something like `flexio_attach`?). That function could then allocate shifters and timers for the child device.

I suppose the reasoning here is that now if the FlexIO is overallocated than initialization will fail for all devices? If child devices requested allocation of FlexIO timers/shifters, than some FlexIO child devices could initialize, even if the FlexIO as a whole was overallocated in the application.